app/assets/javascripts/discourse/app/lib/swipe-events.js

```diff
@@ -0,0 +1,233 @@
+import { isTesting } from "discourse-common/config/environment";
+
+/**
+   Swipe events is a class that allows components to detect and respond to swipe gestures
+   It sets up custom events for swipestart, swipeend, and swipe for beginning swipe, end swipe, and during swipe. Event returns detail.state with swipe state, and the original event.
+
+   Calling preventDefault() on the swipestart event stops swipe and swipeend events for the current gesture. it is re-enabled on future swipe events.
+**/
+export const SWIPE_DISTANCE_THRESHOLD = 50;
+export const SWIPE_VELOCITY_THRESHOLD = 0.12;
+export const MINIMUM_SWIPE_DISTANCE = 5;
+export const MAX_ANIMATION_TIME = 200;
+export default class SwipeEvents {
+  //velocity is pixels per ms
+
+  swipeState = null;
+  animationPending = false;
+
+  constructor(element) {
+    this.element = element;
+  }
+
+  addTouchListeners() {
+    this.touchStart = (e) => {
```

Comment:
Having these method definitions nested inside `addTouchListeners` feels a little weird... I think they be defined cleanly in the body of the class like

```
@bind
touchStart(e){
  ...
}
```

Reply:
Ohh there's a bind DECORATOR, missed that! 🙌